workers/go/resources/tracker.go

```diff
@@ -0,0 +1,192 @@
+package resources
```

Comment:
This whole thing feels like more work than necessary. Let's just use Prometheus to record all this and then we can snapshot the DB when it's done. Maybe another option would work too, but let's not write our own custom metrics format and collection